src/main/java/cz/student/adambrg/UserDao.java

```diff
@@ -0,0 +1,139 @@
+            e.printStackTrace();
+            System.err.println("ERROR while updating the user");
+        }
+    }
+
+    public static final String READ_USER =
+            "SELECT * FROM users WHERE id = ?";
+
+    public User read(int userId, Connection conn) { //metoda pro precteni informaci uzivatele
+
+        try {
+            PreparedStatement statement = conn.prepareStatement(READ_USER);
+            statement.setInt(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                User user = new User();
+                user.setId(resultSet.getInt("id"));
+                user.setUserName(resultSet.getString("username"));
+                user.setEmail(resultSet.getString("email"));
+                user.setPassword(resultSet.getString("password"));
+                System.out.println("Information about user:");
+                return user;
+
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while reading the user");
+        }
+        return null;
+    }
+
+    public static final String DELETE_USER =
+            "DELETE FROM users WHERE id = ?";
+
+    public void delete(int userId, Connection conn) { // metoda pro smazani uzivatele
+        try {
+            PreparedStatement statement = conn.prepareStatement(DELETE_USER);
+            statement.setInt(1, userId);
+           int affectedRows = statement.executeUpdate();
+
+            if (affectedRows > 0) {
+                System.out.println("User with ID: " + userId + " has been successfully removed.");
+            } else {
+                System.out.println("Failed to remove user with ID: " + userId + ".");
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while deleting the user");
+        }
+    }
+
+    public static final String READ_ALL_USERS =
+            "SELECT * FROM users";
+
+    public List<User> findAll(Connection conn) {  //metoda pro zobrazeni vsech uctu
+        try {
+            List<User> users = new ArrayList<>();
+            PreparedStatement statement = conn.prepareStatement(READ_ALL_USERS);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                User user = new User();
+                user.setId(resultSet.getInt("id"));
+                user.setUserName(resultSet.getString("username"));
+                user.setEmail(resultSet.getString("email"));
+                user.setPassword(resultSet.getString("password"));
+                users.add(user);
+            }
+            return users;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("ERROR while reading all the users");
+        }
+        return null;
+    }
+
+
+    public String hashPassword(String password) { // zabezpecovaci metoda hesel
```

Comment:
BUG: hashPassword metoda je interni metoda tridy UserDao, proto by mela byt private.

```suggestion
private String hashPassword(String password) { // zabezpecovaci metoda hesel
```

Reply:
opraveno